test/animation/Animate.progress.spec.tsx

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { describe, expect, it, vi } from 'vitest';
+import { MockProgressAnimationManager } from './MockProgressAnimationManager';
+import { Animate } from '../../src/animation/Animate';
+
+describe('Animate progress', () => {
+  describe('when child is a function', () => {
+    describe('with from, to as objects', () => {
+      it('should call the function child with the current style', async () => {
```

Comment:
This is already quite a nice test isn't it @ckifer. Animations are almost tamed.

Reply:
So much better, thank you 🙌